### PR TITLE
add product page with funtionality

### PR DESCRIPTION
**Pull request by harshkotkar5557**
You can refer this below mentioned for preview:
https://hareshkotkar.tinytake.com/msc/NjY1MTIyM18xOTIxMDUxMg

### CODE REVIEW

**Comment by harshkotkar5557** [comment]
New changes added in same branch

File: src/pages/product/product.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import { products } from '../../data/product.js'
+import { useCart } from '../../context/cart-context'
+import { Cateogery } from '../../data/cateogery'
+import {filterByLowHigh, filterByHighLow, getFilterByPrizeRangeVal, getFilterByRatings, getFilterByCateogry } from './utils'
+function Product() {
+    const [productsList, setProductList] = useState([])
+    const [cateogeryFilters, setCateogeryFilters] = useState([])
+    const [ratingsFilter, setRatingFilter] = useState(null)
+    const [prizeRangeFilter, setPrizeRangeFilter] = useState('High-Low')
+    const [rangeVal, setRangeVal] = useState(120000)
```
Comment (Mahi24-prog) [inline src/pages/product/product.js:11]: use useReducer for multiple state management

**Review by Mahi24-prog — COMMENTED** [review]
You can use useReducer instead of useState for multiple state handle

File: src/pages/product/product.js (hunk 1 above)
Comment (harshkotkar5557) [reply src/pages/product/product.js:11]: For practice purpose I have use usestate but in last my plan is to use usereducer.

**Comment by SumitSaundadkar** [comment]
Good to go after useredcucer otherwise looks good to me.

**Pull request merged**